tests/Unit/Service/Client/ServiceClientTest.php

```diff
@@ -95,16 +95,26 @@ public function testItCanPerformAServiceCallFromEmptyTokenCache(): void
         $this->assertInstanceOf(ResponseInterface::class, $result);
         $this->assertEquals('service response', $result->getBody()->__toString());
 
-        $cacheKey = sprintf('lti1p3-service-client-token-' . $registration->getIdentifier());
+        $cacheKey = sprintf(
+            '%s-%s-%s',
+            'lti1p3-service-client-token',
+            $registration->getIdentifier(),
+            sha1(implode('', []))
+        );
         $this->assertTrue($this->cache->hasItem($cacheKey));
         $this->assertEquals('access_token', $this->cache->getItem($cacheKey)->get());
     }
 
-    public function testItCanPerformAServiceCallFromPopulatedTokenCache(): void
+    public function testItCanPerformAServiceCallFromPopulatedTokenCacheWithoutScopes(): void
     {
         $registration = $this->createTestRegistration();
 
-        $cacheKey = sprintf('lti1p3-service-client-token-' . $registration->getIdentifier());
+        $cacheKey = sprintf(
+            '%s-%s-%s',
+            'lti1p3-service-client-token',
+            $registration->getIdentifier(),
+            sha1(implode('', []))
+        );
```

Comment:
Could extract this to a method since is used 3 times in the test.

Reply:
you'll end up creating a private method that just decorates `sprintf`, wont ease readability / reusability a lot (only the cache prefix can be avoided to be repeated)

worth it ?

Reply:
For me yes, cause this part is not the scope of the test, so IMO it distracts a bit while reading the test. But it is just a suggestion :)

Reply:
if you think it's worth it,then I provided it 👍